buildscripts/linter.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Copyright © 2020 The OpenEBS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+(
+    ## Install golangci-lint binary in the setup to check liniting issues
+    ##
+    ## Running lint checks while building binaries will help more than having as workflows
+    ## For more information: https://golangci-lint.run/usage/install/#local-installation
+    if ! command -v golangci-lint &> /dev/null
+    then
+        echo "Running lint checks has dependency on golangci-lint"
+        echo "Installing golangci-lint tool in the system of version 1.40.1"
+        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "${GOPATH}"/bin v1.40.1
+    fi
+)
+
+# Below script is uesd to check linting errors in code base
+check-linting-issues() {
+    echo "Checking linting issues"
+
+    ## Currently we are running with Default options + other options
+
+    ## Explanation for explicitly mentioned linters:
+    ## exportloopref: checks for pointers to enclosing loop variables
+    ## dupl: Tool for code clone detection
+    ## revive: Drop-in replacement of golint. It allows to enable or disable
+    ##         rules using configuration file.
+    ## bodyclose: checks whether HTTP response body is closed successfully
+    ## goconst: Find repeated strings that could be replaced by a constant
+    ## misspell: Finds commonly misspelled English words in comments
+    ##
+    ## NOTE: Disabling structcheck since it is reporting false positive cases
+    ##       for more information look at https://github.com/golangci/golangci-lint/issues/537
+    golangci-lint run -E exportloopref,dupl,revive,bodyclose,goconst,misspell -D structcheck --timeout 5m0s
```

Comment:
once installation becomes part of bootstrap, we can directly run this in `make golint`. see if we need this script after that?